### PR TITLE
Add Zendesk provider

### LINKED ISSUE #586: Add Zendesk provider
**Is your feature request related to a problem? Please describe.**
[Zendesk ](https://developer.zendesk.com/documentation/ticketing/working-with-oauth/creating-and-using-oauth-tokens-with-the-api/) uses OAuth 2.0 and I have started to implement it as a provider so that the API can be used simply for authentication in .NET projects.

**Describe the solution you'd like**
Add a new provider for Zendesk.

### PR DESCRIPTION
**Pull request by levimatheri**
Adds Zendesk provider

### CODE REVIEW

File: src/AspNet.Security.OAuth.Zendesk/ZendeskAuthenticationOptions.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "id");
+            ClaimActions.MapJsonKey(ClaimTypes.Name, "name");
+            ClaimActions.MapJsonKey(ClaimTypes.Email, "email");
+
+            Scope.Add("read");
+        }
+
+        /// <summary>
+        /// Gets or sets the Zendesk domain name.
+        /// For example: 'glowingwaffle.zendesk.com'.
```
Comment (martincostello) [inline src/AspNet.Security.OAuth.Zendesk/ZendeskAuthenticationOptions.cs]:
```suggestion
        /// For example: <c>glowingwaffle.zendesk.com</c>.
```

**Review by martincostello — APPROVED** [review]

**Comment by martincostello** [comment]
Thanks @levimatheri, this looks great. Can you confirm that you've validated this provider works against a real Zendesk tenant?

**Comment by levimatheri** [comment]
@martincostello Yes, I've validated that it works with a real Zendesk tenant. Will work on resolving the suggestion. Thanks.

**Comment by martincostello** [comment]
Cool - thanks 👍

**Review by martincostello — APPROVED** [review]

**Pull request merged**